### PR TITLE
Allow ScoreTypes to provide their own custom global score message

### PR DESCRIPTION
**Pull request by lerks**
See #612.

---
This change is [Reviewable](https://reviewable.io/reviews/cms-dev/cms/622)

### CODE REVIEW

File: cms/grading/ScoreType.py [hunk 1]
```diff
@@ -73,6 +73,30 @@ def __init__(self, parameters, public_testcases):
         self.max_score, self.max_public_score, self.ranking_headers = \
             self.max_scores()
 
+    def format_score(self, score, max_score, unused_score_details,
```
Comment (stefano-maggiolo) [inline cms/grading/ScoreType.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/s-maggiolo/cms/file/2782001443/issues/source?bid=3167167&fileBranchId=3365636#l76)

Comment (stefano-maggiolo) [inline cms/grading/ScoreType.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/s-maggiolo/cms/file/2782145405/issues/source?bid=3167167&fileBranchId=3365636#l76)

**Comment by stefano-maggiolo** [comment]
:lgtm:

---

Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[cms/server/contest/templates/submission_row.html, line 63 \[r4\]](https://reviewable.io:443/reviews/cms-dev/cms/622#-KKgHYc06rAljgf4lE8x:-KKgHYc06rAljgf4lE8y:-1756797166) ([raw file](https://github.com/cms-dev/cms/blob/01b2573de00a72a464dbdb3aa1b842e18405e8f0/cms/server/contest/templates/submission_row.html#L63)):*
> ```HTML
>     </td>
> 
>     {% if score_type.max_public_score != 0 %}
> ```

You want > 0 also here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/622#-:-KKgImvKPt0ckVPlEXW7:1417088721)*

**Comment by lerks** [comment]
*[cms/server/contest/templates/submission_row.html, line 63 \[r4\]](https://reviewable.io:443/reviews/cms-dev/cms/622#-KKgHYc06rAljgf4lE8x:-KKgb6Jfq8ICzcyg7boX:629122618) ([raw file](https://github.com/cms-dev/cms/blob/01b2573de00a72a464dbdb3aa1b842e18405e8f0/cms/server/contest/templates/submission_row.html#L63)):*
<details><summary><i>Previously, stefano-maggiolo (Stefano Maggiolo) wrote…</i></summary>
> You want > 0 also here?
</details>
Sure, why not! ;)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/622)*

**Comment by lerks** [comment]
Pushed.

**Pull request closed without merging**